service/src/main/java/bio/terra/pipelines/common/utils/PipelineInputTypesEnum.java

```diff
@@ -0,0 +1,112 @@
+package bio.terra.pipelines.common.utils;
+
+import bio.terra.common.exception.ValidationException;
+import java.util.List;
+
+public enum PipelineInputTypesEnum {
+  STRING {
+    @Override
+    public String cast(String fieldName, Object value) {
+      validateNotNullOrEmpty(fieldName, value);
+      if (value instanceof String stringValue) {
+        return stringValue.trim();
+      } else {
+        throw new ValidationException(String.format("%s must be a string", fieldName));
+      }
+    }
+  },
+  INTEGER {
+    @Override
+    public Integer cast(String fieldName, Object value) {
+      validateNotNullOrEmpty(fieldName, value);
+      if (value instanceof Integer integerValue) {
+        return integerValue;
+      } else if (value instanceof String stringValue) {
+        try {
+          return Integer.parseInt(stringValue);
+        } catch (NumberFormatException e) {
+          throw new ValidationException(String.format("%s must be an integer", fieldName));
+        }
+      } else {
+        throw new ValidationException(String.format("%s must be an integer", fieldName));
+      }
+    }
+  },
+  VCF {
+    @Override
+    public String cast(String fieldName, Object value) {
+      validateNotNullOrEmpty(fieldName, value);
+      if (value instanceof String stringValue) {
+        if (!stringValue.trim().endsWith(".vcf.gz")) {
+          throw new ValidationException(
+              String.format("%s must be a path to a VCF file ending in .vcf.gz", fieldName));
+        }
+        return stringValue.trim();
+      } else {
+        throw new ValidationException(String.format("%s must be a string", fieldName));
+      }
+    }
+  },
+  STRING_ARRAY {
+    @Override
+    public List<String> cast(String fieldName, Object value) {
+      validateNotNullOrEmpty(fieldName, value);
+      if (value instanceof List<?> listValue) {
+        try {
+          List<String> stringList =
+              listValue.stream()
+                  .map(itemValue -> STRING.cast(fieldName, itemValue).toString())
```

Comment:
I had to add the `toString()` to the cast output to make SonarLint happy, but I'm not sure why it's necessary, since STRING.cast() returns a String.